lib/src/default_index/revset_engine.rs

```diff
@@ -1092,6 +1092,11 @@ fn build_predicate_fn(
                     .unwrap()
             })
         }
+        RevsetFilterPredicate::HasNoConflict => box_pure_predicate_fn(move |index, pos| {
+            let entry = index.entry_by_pos(pos);
+            let commit = store.get_commit(&entry.commit_id()).unwrap();
+            !commit.has_conflict().unwrap()
+        }),
```

Comment:
Maybe you miss the fact that `.filtered()` is a short for `.intersection(expr)`. You can instead negate the filter expression.
```rust
let has_no_conflict = RevsetExpression::filter(RevsetFilterPredicate::HasConflict).negated();
main_expr.intersection(&has_no_conflict)
```

Reply:
Ah thanks. I indeed didn't read that far. That said, this is no longer needed as I'm going with your advice and ignoring descendants. So I've reverted the commit that added this.